src/cli/rustup_mode.rs

```diff
@@ -1578,6 +1586,17 @@ fn set_profile(cfg: &mut Cfg, m: &ArgMatches<'_>) -> Result<utils::ExitCode> {
     Ok(utils::ExitCode(0))
 }
 
+fn set_auto_self_update(cfg: &mut Cfg, m: &ArgMatches<'_>) -> Result<utils::ExitCode> {
+    if self_update::NEVER_SELF_UPDATE {
+        let mut args = crate::process().args_os();
+        let arg0 = args.next().map(PathBuf::from);
+        let arg0 = arg0.as_ref().and_then(|a| a.to_str());
+        warn!("{} is built with the no-self-update feature: setting auto-self-update will not have any effect.",arg0.ok_or(CLIError::NoExeFilePath)?);
```

Comment:
```suggestion
        let arg0 = arg0.as_ref().and_then(|a| a.to_str());
        let arg0 = arg0.ok_or(CLIError::NoExeFilePath)?;
        warn!("{} is built with the no-self-update feature: setting auto-self-update will not have any effect.",arg0);
```

(I don't mind at all if you don't take this, but I think it'd look neater, the `warn` line is already very long)

Reply:
Fixed. Thanks for your review.